### PR TITLE
improve the error message when pooch is not installed

### PR DESCRIPTION
**Pull request by keewis**
Since it seems difficult to fix the situation using the package manager, this improves the error message by explicitly suggesting to install `pooch`.

- [x] partial fix for #5291
- [ ] Tests added
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: xarray/tutorial.py [hunk 1]
```diff
@@ -77,8 +77,11 @@ def open_dataset(
     """
     try:
         import pooch
-    except ImportError:
-        raise ImportError("using the tutorial data requires pooch")
+    except ImportError as e:
+        raise ImportError(
+            "tutorial.open_dataset depends on pooch to download and manage datasets."
+            " To proceed please install it from PyPI or conda / conda-forge."
+        ) from e
```
Comment (shoyer) [inline xarray/tutorial.py:84]: I'm not entirely sure this is better. People already used a package manager (of some form) to install xarray, so mentioning pypi/conda is not necessarily helpful.

Comment (keewis) [reply xarray/tutorial.py:84]: fair enough. Would dropping that be better: "To proceed please install pooch."

Comment (shoyer) [reply xarray/tutorial.py:84]: yes!

**Review by max-sixty — APPROVED** [review]
Great!

**Review by shoyer — APPROVED** [review]

**Comment by keewis** [comment]
thanks for the reviews

**Pull request merged**